web/crux-ui/src/models/container-merge.ts

```diff
@@ -65,6 +65,68 @@ const mergeUniqueKeys = <T extends UniqueKey>(strong: T[], weak: T[]): T[] => {
   return [...strong, ...missing]
 }
 
+const mergeUniqueKeyValues = <T extends UniqueKeyValue>(strong: T[], weak: T[]): T[] => {
+  if (!strong) {
+    return weak ?? null
+  }
+
+  if (!weak) {
+    return strong
+  }
+
+  const overriddenKeys = new Set(strong.map(it => it.key))
+  const overriddenValues = new Set(strong.filter(it => it.value).map(it => it.key))
+  const weakValues = new Set(
+    weak.filter(it => it.value && !overriddenValues.has(it.key) && overriddenKeys.has(it.key)).map(it => it.key),
+  )
+
+  const missing = weak.filter(it => weakValues.has(it.key) || !overriddenKeys.has(it.key))
+  const overriden = strong.filter(it => !weakValues.has(it.key))
+
+  return [...overriden, ...missing]
+}
+
+const mergePorts = (strong: Port[], weak: Port[]): Port[] => {
+  if (!strong) {
+    return weak ?? null
+  }
+
+  if (!weak) {
+    return strong
+  }
+
+  const missing = weak.filter(w => !strong.find(it => it.internal === w.internal))
+  return [...strong, ...missing]
+}
+
+const mergePortRanges = (strong: ContainerPortRange[], weak: ContainerPortRange[]): ContainerPortRange[] => {
+  if (!strong) {
+    return weak ?? null
+  }
+
+  if (!weak) {
+    return strong
+  }
+
+  const missing = weak.filter(
+    w => !strong.find(it => rangesOverlap(w.internal, it.internal) || rangesOverlap(w.external, it.external)),
+  )
+  return [...strong, ...missing]
+}
+
+const mergeVolumes = (strong: Volume[], weak: Volume[]): Volume[] => {
+  if (!strong) {
+    return weak ?? null
+  }
+
+  if (!weak) {
+    return strong
+  }
+
+  const missing = weak.filter(w => !strong.find(it => it.path === w.path || it.name === w.name))
```

Comment:
Fully duplicated with mergePorts, no? Also, could you add some descriptions for these functions, please?

Reply:
(I know, this is not new code just boy scout rule) 🌵